### PR TITLE
fix(components): message Cannot set properties of null(setting visible)

### LINKED ISSUE #8796: [Component] [message] ElMessage close error
Bug Type: **`Component`**

## Environment

- Vue Version: `3.2.31`
- Element Plus Version: `2.2.8`
- Browser / OS: `'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/99.0.4844.51 Safari/537.36'`
- Build Tool: `Vue CLI`

### Related Component

- `el-message`

### Reproduction Link

[Link]()

### Steps to reproduce

`ElMessage.error('主要信息页签存在必填项未填写或格式不正确，请检查')`

### What is Expected?

Auto Close

### What is actually happening?

Cannot close and throw error

### Additional comments

All ElMessage close will throw error in my pro

image
image

### PR DESCRIPTION
**Pull request by StephenKe**
Please make sure these boxes are checked before submitting your PR, thank you!

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [x] Add some descriptions and refer to relative issues for your PR.
fix #8796

### CODE REVIEW

**Comment by hackycy** [comment]
我也遇到了这个问题，但我无法查找出该问题产生的原因= =

**Comment by jesse-tang-93** [comment]
我也遇到了这个问题，一直没排查到问题

**Review by chenxch — APPROVED** [review]

**Comment by chenxch** [comment]
> I met this issue right now ,will you guys merge this soon?

If it is more urgent, you can upgrade the vue version first, which should be updated in the recent version.

File: packages/components/message/src/instance.ts [hunk 1]
```diff
@@ -26,5 +26,5 @@ export const getInstance = (id: string) => {
 export const getLastOffset = (id: string): number => {
   const { prev } = getInstance(id)
   if (!prev) return 0
-  return prev.vm.exposeProxy!.bottom
+  return prev.vm?.exposeProxy?.bottom || 0
```
Comment (sxzz) [inline packages/components/message/src/instance.ts:29]:
```suggestion
  return prev.vm.exposeProxy?.bottom || 0
```

**Review by sxzz — COMMENTED** [review]
Please revert permission of file
`100644 → 100755`

File: packages/components/message/src/instance.ts (hunk 1 above)
Comment (chenxch) [reply packages/components/message/src/instance.ts:29]: Then the judgment in another file should also be adjusted.

**Comment by yinshipeng** [comment]
解决了，我把vue升级到3.2.36以上就可以了

**Comment by sjzgit** [comment]
> 解决了，我把vue升级到3.2.36以上就可以了

vue 3.2.37 element-plus 2.2.9 依然报错

**Comment by XiaoBaiClassmate** [comment]
vue:3.2.45
element-plus:2.2.28
pinia:2.0.29

配合pinia state使用的时候也会报该错误。

**Pull request closed without merging**